spec/ParseUser.spec.js

```diff
@@ -4031,4 +4031,30 @@ describe('Security Advisory GHSA-8w3j-g983-8jh5', function () {
     const user = await query.get('1234ABCDEF', { useMasterKey: true });
     expect(user.get('authData')).toEqual({ custom: { id: 'linkedID' } });
   });
+
+  describe('issue #6641', () => {
+    fit('succes to login when masterkey and instalationId is provided and when password is not provided', done => {
+      const user = new Parse.User();
+      user
+        .save({
+          username: 'yolo',
+          password: 'yolopass',
+          email: 'yo@lo.com',
+        })
+        .then(async savedUser => {
+          const installationId = '123456789';
+          const query = new Parse.Query(Parse.User);
+          const result = await query.get(savedUser.id, {
+            useMasterKey: true,
+            installationId: installationId,
+          });
+          return await result.logIn({ useMasterKey: true, installationId: installationId });
+        })
+        .then(() => done())
+        .catch(() => {
+          fail();
```

Comment:
I think we don't need any fail(), done(), these should be avoided when possible.

Reply:
the function never end if I don't use `done()`

Reply:
- for exceptions use expect().toThrow()
- for promise rejection a use expect().toBeRejected()